frame/staking/src/migrations.rs

```diff
@@ -288,214 +287,217 @@ pub mod v10 {
-		assert!(
-			CounterForValidators::<T>::get().unwrap().is_zero(),
-			"CounterForValidators already set."
-		);
-		assert!(
-			CounterForNominators::<T>::get().unwrap().is_zero(),
-			"CounterForNominators already set."
-		);
-		assert!(Validators::<T>::count().is_zero(), "Validators already set.");
-		assert!(Nominators::<T>::count().is_zero(), "Nominators already set.");
-		assert!(StorageVersion::<T>::get() == ObsoleteReleases::V6_0_0);
-		Ok(())
-	}
-
-	pub fn migrate<T: Config>() -> Weight {
-		log!(info, "Migrating staking to ObsoleteReleases::V7_0_0");
-		let validator_count = Validators::<T>::iter().count() as u32;
-		let nominator_count = Nominators::<T>::iter().count() as u32;
-
-		CounterForValidators::<T>::put(validator_count);
-		CounterForNominators::<T>::put(nominator_count);
-
-		StorageVersion::<T>::put(ObsoleteReleases::V7_0_0);
-		log!(info, "Completed staking migration to ObsoleteReleases::V7_0_0");
-
-		T::DbWeight::get().reads_writes(validator_count.saturating_add(nominator_count).into(), 2)
-	}
-}
-
-pub mod v6 {
-	use super::*;
-	use frame_support::{storage_alias, traits::Get, weights::Weight};
-
-	// NOTE: value type doesn't matter, we just set it to () here.
-	#[storage_alias]
-	type SnapshotValidators<T: Config> = StorageValue<Pallet<T>, ()>;
-	#[storage_alias]
-	type SnapshotNominators<T: Config> = StorageValue<Pallet<T>, ()>;
-	#[storage_alias]
-	type QueuedElected<T: Config> = StorageValue<Pallet<T>, ()>;
-	#[storage_alias]
-	type QueuedScore<T: Config> = StorageValue<Pallet<T>, ()>;
-	#[storage_alias]
-	type EraElectionStatus<T: Config> = StorageValue<Pallet<T>, ()>;
-	#[storage_alias]
-	type IsCurrentSessionFinal<T: Config> = StorageValue<Pallet<T>, ()>;
-
-	/// check to execute prior to migration.
-	pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
-		// these may or may not exist.
-		log!(info, "SnapshotValidators.exits()? {:?}", SnapshotValidators::<T>::exists());
-		log!(info, "SnapshotNominators.exits()? {:?}", SnapshotNominators::<T>::exists());
-		log!(info, "QueuedElected.exits()? {:?}", QueuedElected::<T>::exists());
-		log!(info, "QueuedScore.exits()? {:?}", QueuedScore::<T>::exists());
-		// these must exist.
-		assert!(
-			IsCurrentSessionFinal::<T>::exists(),
-			"IsCurrentSessionFinal storage item not found!"
-		);
-		assert!(EraElectionStatus::<T>::exists(), "EraElectionStatus storage item not found!");
-		Ok(())
-	}
-
-	/// Migrate storage to v6.
-	pub fn migrate<T: Config>() -> Weight {
-		log!(info, "Migrating staking to ObsoleteReleases::V6_0_0");
-
-		SnapshotValidators::<T>::kill();
-		SnapshotNominators::<T>::kill();
-		QueuedElected::<T>::kill();
-		QueuedScore::<T>::kill();
-		EraElectionStatus::<T>::kill();
-		IsCurrentSessionFinal::<T>::kill();
-
-		StorageVersion::<T>::put(ObsoleteReleases::V6_0_0);
-
-		log!(info, "Done.");
-		T::DbWeight::get().writes(6 + 1)
-	}
-}
+// These migrations have been commented out as they are no longer compatible with the current
```

Comment:
why? And, let's try rewriting them in a compatible way? https://github.com/paritytech/substrate/issues/10308

Reply:
We can do that, however that feels like unnecessary maintenance. If we want to be thorough about it - we'd also need to test them on live data of a particular block. That is a lot of extra work for any change that breaks one or more migrations. One might argue that if you still want to run them - just migrate gradually using the old releases? Do we really want to start introducing this burden to seemingly unrelated PRs?

I'm afraid that's going to put unnecessary strain on lots of future PRs. And without testing those migrations are basically useless as they are likely to fail, as we can only test them manually.